src/com/google/ProfilingReport.java

```diff
@@ -0,0 +1,249 @@
+
+  int totalReportsParsed;
+  float percentOfFunctionsExecuted;
+  float percentOfBranchesExecuted;
+
+  List<FileProfilingResult> result;
+
+
+  private ProfilingReport(int totalReportsParsed, List<FileProfilingResult> result) {
+    this.totalReportsParsed = totalReportsParsed;
+    this.result = result;
+
+    calculateProfilingReportPercents();
+  }
+
+  private static float roundFloat2Decimals(float num) {
+    return (Math.round(num * 10000) / (float) 100);
+  }
+
+  /**
+   * This function takes the instrumentationMapping and reports sent by the instrumented production
+   * code and creates a list of ProfilingResultsByFile with the aggregated information. It will
+   * average the ProfilingData provided by allInstrumentationReports and also calculate a percentage
+   * of how often a parameter is encounter.
+   *
+   * @param instrumentationMapping    The instrumentationMapping generated by the Production
+   *                                  Instrumentation pass
+   * @param allInstrumentationReports A list off all reports sent by the instrumented production
+   *                                  code
+   * @return A list of ProfilingResultsByFile where each element is the profiling result for a
+   * different source file.
+   */
+  public static ProfilingReport createProfilingReport(
+      InstrumentationMapping instrumentationMapping,
+      List<Map<String, ProfilingData>> allInstrumentationReports) {
+
+    List<FileProfilingResult> result = new ArrayList<>();
+
+    // Iterate over all fileNames since that is what we are grouping by.
+    for (String fileName : instrumentationMapping.fileNames) {
+
+      // Get all instrumentation parameters where the source file is fileName.
+      List<String> instrumentationPointsPerFile = instrumentationMapping.getAllMatchingValues((s) ->
+          fileName.equals(instrumentationMapping.getFileName(s))
+      );
+
+      Map<String, List<ProfilingResult>> profilingDataPerFunction = new HashMap<>();
+
+      for (String param : instrumentationPointsPerFile) {
+        String functionName = instrumentationMapping.getFunctionName(param);
+
+        float totalInstrumentationPointsExecuted = 0;
+        int totalFrequency = 0;
+        int instrumentationPointsCounter = 0;
+
+        // For each param, iterate over allInstrumentationReports and check if param is present.
+        // If it is, we will average the data, otherwise we will add to the average as if it is 0.
+        for (Map<String, ProfilingData> instrumentationData : allInstrumentationReports) {
+          ProfilingData profilingData = instrumentationData.get(param);
+          instrumentationPointsCounter++;
+          if (profilingData != null) {
+            // if executed, add 1 to executionAverage, and 0 otherwise.
+            totalFrequency += profilingData.frequency;
+            totalInstrumentationPointsExecuted++;
+          }
+        }
+
+        // Round the executionAverage to 2 decimal places for simplicity.
```

Comment:
Comment seems out of date

Reply:
Updated to code instead of comment since name felt vague to me.